src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,108 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTORY = 0.75;
+    private static final int CAPACITY_MULTIPLIER = 2;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+        threshold = (int) (DEFAULT_LOAD_FACTORY * DEFAULT_INITIAL_CAPACITY);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> node = new Node<>(key, value);
+        resize();
+        if (table[getIndex(key)] == null) {
+            table[getIndex(key)] = node;
+            size++;
+        } else {
+            Node<K, V> oldNode = table[getIndex(key)];
+            oldNode = getRightNode(oldNode, key);
+            if (checkEquals(oldNode.key, key)) {
+                oldNode.value = node.value;
+                return;
+            }
+            oldNode.next = node;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        V value = null;
```

Comment:
follow IDEA's hints and suggestions
![image](https://github.com/mate-academy/jv-hash-map/assets/49767049/f434332f-4b62-43d8-a6c0-ef06980271bb)